app/vmalert/group.go

```diff
@@ -326,6 +330,10 @@ func (g *Group) start(ctx context.Context, nts func() []notifier.Notifier, rw *r
 		}
 
 		resolveDuration := getResolveDuration(g.Interval, *resendDelay, *maxResolveDuration)
+		if g.queryTimeAlignment == nil || *g.queryTimeAlignment {
+			// see https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5049
```

Comment:
nit: would be nice to mention https://github.com/VictoriaMetrics/VictoriaMetrics/pull/1257 as well

Reply:
Added.